experimental/dds/tree2/src/test/feature-libraries/editable-tree-2/objectFactory.spec.ts

```diff
@@ -179,11 +187,223 @@ describe("SharedTreeObject factories", () => {
+				return this.objects;
+			}
+
+			public createComboTree(): ComboParent {
+				return this.createComboParent();
+			}
+
+			private createComboParent(): ComboParent {
+				const object = comboParent.create({
+					id: this.nextId++,
+					object: this.createComboChild(),
+					objectList: [this.createComboChild(), this.createComboChild()],
+					objectMap: new Map([
+						["A", this.createComboChild()],
+						["B", this.createComboChild()],
+					]),
+					listList: [
+						[this.createComboChild(), this.createComboChild()],
+						[this.createComboChild(), this.createComboChild()],
+					],
+					listMap: new Map([
+						["A", [this.createComboChild(), this.createComboChild()]],
+						["B", [this.createComboChild(), this.createComboChild()]],
+					]),
+					mapList: [
+						new Map([
+							["A", this.createComboChild()],
+							["B", this.createComboChild()],
+						]),
+						new Map([
+							["A", this.createComboChild()],
+							["B", this.createComboChild()],
+						]),
+					],
+					mapMap: new Map([
+						[
+							"A",
+							new Map([
+								["A", this.createComboChild()],
+								["B", this.createComboChild()],
+							]),
+						],
+						[
+							"B",
+							new Map([
+								["A", this.createComboChild()],
+								["B", this.createComboChild()],
+							]),
+						],
+					]),
+				});
+				this.objects.push(object);
+				return object;
+			}
+
+			private createComboChild(): ComboChild {
+				const object = comboChild.create({
+					id: this.nextId++,
+					object: this.createComboLeaf(),
+					objectList: [this.createComboLeaf(), this.createComboLeaf()],
+					objectMap: new Map([
+						["A", this.createComboLeaf()],
+						["B", this.createComboLeaf()],
+					]),
+					listList: [
+						[this.createComboLeaf(), this.createComboLeaf()],
+						[this.createComboLeaf(), this.createComboLeaf()],
+					],
+					listMap: new Map([
+						["A", [this.createComboLeaf(), this.createComboLeaf()]],
+						["B", [this.createComboLeaf(), this.createComboLeaf()]],
+					]),
+					mapList: [
+						new Map([
+							["A", this.createComboLeaf()],
+							["B", this.createComboLeaf()],
+						]),
+						new Map([
+							["A", this.createComboLeaf()],
+							["B", this.createComboLeaf()],
+						]),
+					],
+					mapMap: new Map([
+						[
+							"A",
+							new Map([
+								["A", this.createComboLeaf()],
+								["B", this.createComboLeaf()],
+							]),
+						],
+						[
+							"B",
+							new Map([
+								["A", this.createComboLeaf()],
+								["B", this.createComboLeaf()],
+							]),
+						],
+					]),
+				});
+				this.objects.push(object);
+				return object;
+			}
+
+			private createComboLeaf(): ComboLeaf {
+				const object = comboLeaf.create({ id: this.nextId++ });
+				this.objects.push(object);
+				return object;
+			}
+		}
+
+		// This test checks that every proxy in the combo tree gets hydrated after insertion
+		const view = createTreeView2(comboSchema, { root: undefined });
```

Comment:
Consider making its? If not, document that this test is uber and can have individual scenarios added on top later.